File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        l = 0
        r = len(nums)

        while l < r:
            if nums[l] == val:
                nums[l] = nums[r - 1]
                r = r - 1
            else:
                l = l + 1

        return r

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<vector<int>> bucket(nums.size()+1);
        unordered_map<int, int> freq;
        vector<int> topknums{};

        for(const auto& num:nums){
            freq[num] += 1;
        }

        for(auto [num,f]: freq){
            bucket[f].push_back(num);
        }

        int numssize = nums.size();
        int currk = 0;
        
        for(int i = nums.size(); i >=0; i--){
            for(int bi=0; bi < bucket[i].size(); bi++){
                
                if(topknums.size() == k) return topknums;

                topknums.push_back(bucket[i][bi]);
            }
        }
        return topknums;

    }
};
